File: Easy/Challenge #001/Python/code.py
#!/usr/bin/python3

"""
Solution for Easy Challenge #001
Author: Keydrain
Date:   03/25/15

Includes 'Extra Credit' file management
Once the out.log file has been created, it must be deleted to be changed. 
"""

import sys

FILE = "out.log"
try:
	f = open(FILE, 'r')
	print(f.read())
except:
	f = open(FILE, 'w+')
	name = input("Please enter your name: ")
	age = input("Please enter your age: ")
	username = input("Please enter your reddit username: ") 
	print("Your name is %s, you are %s years old, and your username is %s\n" % (name, age, username))
	f.write("Your name is %s, you are %s years old, and your username is %s\n" % (name, age, username))

File: Easy/Challenge #001/C/code.c
/*
	Solution for Easy Challenge #001
	Author: Keydrain
	Date:   03/24/15

	Includes 'Extra Credit' file management
	Once the out.log file has been created, it must be deleted to be changed. 
*/


#include <stdio.h> 

int main(void){

	FILE *input, *output;
	char logfile[] = "out.log";
	int c;

	input = fopen(logfile, "r");
	if (input == NULL){

		char name[64];
		int length = 0;
		int age = 0;
		char username[64];
	
		printf("Please enter your name: ");
		scanf ("%[^\n]%*c", name); /* Regex in scanf for C */
		printf("Please enter your age: ");
		scanf("%d", &age);
		printf("Please enter your reddit username: ");
		rewind(stdin); /* This gem puts the reader back at the beginning of stdin. */
		scanf ("%[^\n]%*c", username); /* Regex in scanf for C */
		printf("Your name is %s, you are %d years old, and your username is %s\n", name, age, username);
		
		output = fopen(logfile, "w");
		fprintf(output, "Your name is %s, you are %d years old, and your username is %s\n", name, age, username);
		fclose(output);
		return(0);

	} else {

		while ((c = getc(input)) != EOF) {
			putchar(c);
		}
		fclose(input);
		return(0);

	}
}